modules/navier_stokes/src/fvkernels/INSFVEnergyAdvection.C

```diff
@@ -36,9 +36,14 @@ ADReal
 INSFVEnergyAdvection::computeQpResidual()
 {
   const auto v = velocity();
+  const auto & limiter_time = _subproblem.isTransient()
+                                  ? Moose::StateArg(1, Moose::SolutionIterationType::Time)
+                                  : Moose::StateArg(1, Moose::SolutionIterationType::Nonlinear);
```

Comment:
Because I don't have an old time step and if I limit implicitly the whole limiting theory breaks and the condition number of the Jacobian is really bad. Note that this works for both Newton steay and nonlinear SIMPLE. In brief, we wants a limiter that converges as good as upwind while still limiting